tensorboard/webapp/metrics/views/card_renderer/scalar_card_component.ts

```diff
@@ -242,13 +242,20 @@ export class ScalarCardComponent<Downloader> {
       end: null,
     };
 
-    if (this.linkedTimeSelection !== null) {
-      this.onLinkedTimeSelectionChanged.emit({
+    // Updates linked time selection regardless of linked time enableness.
+    // This enables the app to use the lastest step selector update as linked time selection
+    this.onLinkedTimeSelectionChanged.emit({
+      timeSelection,
+      // When linkedTime is not enabled, we only update time selection state.
+      ...(this.linkedTimeSelection !== null && {affordance}),
+    });
+
+    // Fires onStepSelectorTimeSelectionChanged for internal analysis
+    if (this.linkedTimeSelection === null) {
+      this.onStepSelectorTimeSelectionChanged.emit({
```

Comment:
Invoking Stephan: You've introduced composite actions (firing two ngrx actions from the same user action to modify two parts of state), which is a smell in our code base.  Ideally there is a single ngrx action to represent one user action.

Is it possible to rewrite this as a single action? It would require downstream reducers and effects to determine whether linked time is enabled or not and handle the action based on that.